libs/sdk-core/src/breez_services.rs

```diff
@@ -1323,9 +1323,16 @@ impl OpeningFeeParamsMenu {
         ensure!(!is_empty, "Validation failed: no fee params provided");
 
         // opening_fee_params_menu fees must be in ascending order
-        // Sort by key "tuple (min_msat, proportional)", which is (primary sorting key, secondary sorting key)
         let is_ordered = self.vals.windows(2).all(|ofp| {
-            (ofp[0].min_msat, ofp[0].proportional) <= (ofp[1].min_msat, ofp[1].proportional)
+            let larger_min_msat_fee = ofp[0].min_msat < ofp[1].min_msat;
+            let equal_min_msat_fee = ofp[0].min_msat == ofp[1].min_msat;
+
+            let larger_proportional = ofp[0].proportional < ofp[1].proportional;
+            let equal_proportional = ofp[0].proportional == ofp[1].proportional;
+
+            (larger_min_msat_fee && equal_proportional)
+                || (equal_min_msat_fee && larger_proportional)
+                || (larger_min_msat_fee && larger_proportional)
```

Comment:
This is semantically the exact condition on spec but doesn't it boil down to each succeeding item must have either a larger min_msat or a larger proportional?

i.e.,
```
let is_ordered = opening_fee_params_menu.windows(2).all(|ofp| {
    ofp[0].min_msat < ofp[1].min_msat || ofp[0].proportional < ofp[1].proportional
});
```